From c104da44ef84a8a60b62d94241b869079d3e1fd2 Mon Sep 17 00:00:00 2001
From: Abhay Krishna Arunachalam <arnchlm@amazon.com>
Date: Mon, 11 Nov 2024 17:25:26 -0800
Subject: [PATCH 6/7] Use docker_tag file to fetch Kubernetes source version

Kind v0.25.0 introduced a change to use the `version` file in the Kubernetes server binaries tarball to
fetch the Kubernetes source version. But the server tarball from EKS Distro does not include this file.
So we are reverting that change to keep the previous behavior of using the docker_tag file as the source
of truth for the Kubernetes version

Signed-off-by: Abhay Krishna Arunachalam <arnchlm@amazon.com>
---
 pkg/build/nodeimage/internal/kube/builder_remote.go  | 2 +-
 pkg/build/nodeimage/internal/kube/builder_tarball.go | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/pkg/build/nodeimage/internal/kube/builder_remote.go b/pkg/build/nodeimage/internal/kube/builder_remote.go
index c76e4517..901b1675 100644
--- a/pkg/build/nodeimage/internal/kube/builder_remote.go
+++ b/pkg/build/nodeimage/internal/kube/builder_remote.go
@@ -78,7 +78,7 @@ func (b *remoteBuilder) Build() (Bits, error) {
 	}
 
 	binDir := filepath.Join(tmpDir, "kubernetes/server/bin")
-	contents, err := os.ReadFile(filepath.Join(tmpDir, "kubernetes/version"))
+	contents, err := os.ReadFile(filepath.Join(binDir, "kube-apiserver.docker_tag"))
 	// fallback for Kubernetes < v1.31 which doesn't have the version file
 	// this approach only works for release tags as the format happens to match
 	// for pre-release builds the docker tag is mangled and not valid semver
diff --git a/pkg/build/nodeimage/internal/kube/builder_tarball.go b/pkg/build/nodeimage/internal/kube/builder_tarball.go
index 58c2a58c..23547a84 100644
--- a/pkg/build/nodeimage/internal/kube/builder_tarball.go
+++ b/pkg/build/nodeimage/internal/kube/builder_tarball.go
@@ -59,7 +59,7 @@ func (b *directoryBuilder) Build() (Bits, error) {
 	}
 
 	binDir := filepath.Join(tmpDir, "kubernetes/server/bin")
-	contents, err := os.ReadFile(filepath.Join(tmpDir, "kubernetes/version"))
+	contents, err := os.ReadFile(filepath.Join(binDir, "kube-apiserver.docker_tag"))
 	// fallback for Kubernetes < v1.31 which doesn't have the version file
 	// this approach only works for release tags as the format happens to match
 	// for pre-release builds the docker tag is mangled and not valid semver
-- 
2.50.1 (Apple Git-155)

